### PR TITLE
Add inference

### LINKED ISSUE #7: Run Inference
## Detailed Description
Given a trained model run inference with the follow objectives
- Produce csv ready to hand over to client. 
- run metrics on test set, understand the MAE

## Context
Useufl to run on test set, so we are not running on training set

## Possible Implementation
CSV shouwl ahve the following columns
- ocf_id
- target_time_utc, with timezone to be clear. The time the forecast is for. 
- creation_time_utc, the time at which the forecast was made
- solar_generation_forecast_kw, the forecast value.
- solar_generation_actual_kw, the actual value.

### PR DESCRIPTION
**Pull request by carderne**
## Description
Resolves #7 

Add inference, post-processing, metrics.

## Checklist:
- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: infer.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    pipeline: Path, module: str, ckpt_path: Path, batches: int
+) -> tuple[np.ndarray, Batch, int]:
+    """Run inference and return y_hat and batches.
+
+    Returns:
+        y_hat: np.ndarray shape [num_preds, forward_time_steps]
+        truth: dict of np.ndarray, shaped [num_preds, ...]
+        pv_t0_idx: int
+    """
+    pv_data_pipeline = simple_pv_datapipe(pipeline, tag="test")
```
Comment (peterdudfield) [inline infer.py]: rename name pipeline to `configuration_file`?
